### DISCUSSION TITLE
Text file import structure?

### DISCUSSION THREAD

**Discussion by janaxhell** [sbondCo/Watcharr#682] [Q&A]
I've just deployed Watcharr. I started tracking my movies by hand years before services like Trakt existed, so all I have is a database from which I can export data to a txt file. I see txt import is among the options, but there is no explanation about what structure to use. 
What does Watcharr expect from a txt file?

**Comment by IRHM** [comment]
Hi @janaxhell, apologies this isn't documented.

It should accept a list of one movie/show per line, like so:

```
Show name
Movie name
```

With optional support for adding the content release year:

```
Show name (2019)
Movie name (2020)
Movie name without a year
```

p.s. This is how I used to manage my list before watcharr, just a text file like so. I remember importing mine, it can be a bit tedious this way since it's very likely to get multiple results for at least a few shows/movies, which require manual intervention (it should only automatically match content if there is only one result or there is only one perfect match in the results).

Anyways, god speed. Please let me know if I can help further with this or anything else (in a new discussion or issue).

**Reply by janaxhell** [reply]
Thanks, what about language? I'm from Italy, so all my movies are catalogued with both original and Italian AKA (if existing). Italian (or any other non-ENG country) native films have title in their original language and it's the AKA that's in English: will the parser understand them? I'm asking because I often found applications that tended to be anglocentric (TMDB itself often struggles with non US/UK content on Filebot). 
EDIT
Also, will it understand trailing articles, like "Movie Title, The (1987)" ?

**Reply by IRHM** [reply]
Hi @janaxhell, unfortunately it uses TMDB for the search, you could try searching for some of the content through watcharrs searchbar (or tmdb) in the specific language, if it pops up, then it should be working.

Watcharr just sends the full movie or show name to tmdb (exactly how it reads it from the text file), so unfortunately if it doesn't work with tmdb, it likely won't here.

If you have some trouble with certain movie/show names, you could reply with some example movie/shows and I could give it a try and see if theres anything we can do with the TMDB api to return results any better (or in a specific language set in your user profile). I'm not 100% sure theres anything I can do, but I will try to look into it.

**Reply by janaxhell** [reply]
Hello again, I've been busy for a while and just had time to try a txt list.
- the parser doesn't recognize accented letters like àèìòù and skips titles
- the parser doesn't understand titles with trailing article, like Thing, The (1982)
- the multiple results dialog doesn't show the year of the ambiguous title and since the list is blurred, one cannot tell which version is
- the parser doesn't understand very common words like - Theatrical, - Director's cut, - Remaster, - cut, - uncut, - rated, and so on

Besides showing the year in the multiple choice dialog, I suggest to add an Edit button to allow to fix a title while parsing and/or add an option to bypass all problematic titles. I have a list of about 7500 titles and it's not possible to sit for days choosing the right title, otherwise the program halts waiting for an input.

**Reply by dannyyy** [reply]
For shows, what's the format to consider seasons as episodes? Different from a movie, a show is mostly not either watched or unwachtd.
